output-comparator/src/main/java/org/mobilitydata/gtfsvalidator/outputcomparator/cli/Main.java

```diff
@@ -0,0 +1,188 @@
+          continue;
+        }
+        int newErrorCount;
+        ValidationReport referenceReport = ValidationReport.fromPath(referenceReportPath);
+        ValidationReport latestReport = ValidationReport.fromPath(latestReportPath);
+        if (referenceReport.hasSameErrorCodes(latestReport)) {
+          continue;
+        }
+
+        for (String noticeCode : referenceReport.getNewErrorsListing(latestReport)) {
+          NoticeStat noticeStat =
+              acceptanceTestReportMap.getOrDefault(noticeCode, new NoticeStat());
+          acceptanceTestReportMap.putIfAbsent(noticeCode, noticeStat);
+          noticeStat.update(
+              file.getName(),
+              latestReport.getNoticeByCode(noticeCode).getTotalNotices(),
+              urlsAsString);
+        }
+
+        newErrorCount = referenceReport.getNewErrorCount(latestReport);
+        if (newErrorCount >= args.getNewErrorThreshold()) {
+          ++badDatasetCount;
+        }
+      }
+      exportAcceptanceTestReport(generateAcceptanceTestReport(acceptanceTestReportMap), args);
+      checkRuleValidity(
+          badDatasetCount, totalDatasetCount, args.getPercentInvalidDatasetsThreshold());
+    } catch (IOException e) {
+      logger.atSevere().withCause(e);
+      System.exit(IO_EXCEPTION_EXIT_CODE);
+    }
+  }
+
+  /**
+   * Generates acceptance test report.
+   *
+   * @param acceptanceTestReportData acceptance test data (mapped by sourceId, {@code NoticeStat})
+   * @return the {@code JsonObject} representation of the acceptance test report
+   */
+  @VisibleForTesting
+  public static JsonObject generateAcceptanceTestReport(
+      Map<String, NoticeStat> acceptanceTestReportData) {
+    JsonObject root = new JsonObject();
+    JsonArray jsonNotices = new JsonArray();
+    root.add("newErrors", jsonNotices);
+
+    for (String noticeCode : acceptanceTestReportData.keySet()) {
+      JsonObject noticeStatJson = new JsonObject();
+      jsonNotices.add(noticeStatJson);
+      noticeStatJson.add(noticeCode, acceptanceTestReportData.get(noticeCode).toJson());
+    }
+    return root;
+  }
+
+  /**
+   * Exports acceptance test reports.
+   *
+   * @param acceptanceTestReportData the JSON representation of the acceptance test report
+   * @param arguments the {@code Arguments} used to retrieve the path to save files
+   */
+  private static void exportAcceptanceTestReport(
+      JsonObject acceptanceTestReportData, Arguments arguments) {
+    new File(arguments.getOutputBase()).mkdirs();
+    try {
+      Files.write(
+          Paths.get(arguments.getOutputBase(), ACCEPTANCE_REPORT_JSON),
+          GSON.toJson(acceptanceTestReportData).getBytes(StandardCharsets.UTF_8));
+    } catch (IOException e) {
+      logger.atSevere().withCause(e).log("Cannot store acceptance test report file");
+    }
+  }
+
+  /**
+   * Exits on non-zero code 2 if the ratio badDatasetCount/totalDatasetCount is greater than or
```

Comment:
I would reference the constant name `INVALID_NEW_RULE_EXIT_CODE` here and not the value in case they change in the future.